apps/OpenSpace/ext/launcher/src/profile/horizonsdialog.cpp

```diff
@@ -0,0 +1,1173 @@
+            _chooseObserverCombo->show();
+            break;
+        }
+        case openspace::HorizonsFile::ResultCode::ErrorNoTarget:
+            appendLog(
+                fmt::format("No match was found for target '{}'", _targetName),
+                HorizonsDialog::LogLevel::Error
+            );
+            appendLog(
+                fmt::format("Try to use '{}*' as target to search for possible matches.",
+                    _targetName),
+                HorizonsDialog::LogLevel::Info
+            );
+            styleLabel(_targetLabel, true);
+            break;
+        case openspace::HorizonsFile::ResultCode::MultipleTarget: {
+            // Case Small Bodies:
+            // Line before data: Matching small-bodies
+            // Format: Record #, Epoch-yr, >MATCH DESIG<, Primary Desig, Name
+            // Line after data:
+            // (X matches. To SELECT, enter record # (integer), followed by semi-colon.)
+
+            // Case Major Bodies:
+            // Line before data: Multiple major-bodies match string "X*"
+            // Format: ID#, Name, Designation, IAU/aliases/other
+            // Line after data: Number of matches =  X. Use ID# to make unique selection.
+
+            appendLog(
+                fmt::format("Multiple matches were found for target '{}'", _targetName),
+                HorizonsDialog::LogLevel::Warning
+            );
+            styleLabel(_targetLabel, true);
+
+            std::vector<std::string> matchingTargets =
+                _horizonsFile.parseMatches("Name", "matches", ">MATCH NAME<");
+            if (matchingTargets.empty()) {
+                appendLog("Could not parse the matching targets",
+                    HorizonsDialog::LogLevel::Error
+                );
+                if (!_latestHorizonsError.empty()) {
+                    appendLog(
+                        fmt::format("Latest Horizons error: {}", _latestHorizonsError),
+                        LogLevel::Error
+                    );
+                }
+                return false;
+            }
+            _chooseTargetCombo->clear();
+            for (const std::string& target : matchingTargets) {
+                _chooseTargetCombo->addItem(
+                    QString::fromStdString(target),
+                    findId(target)
+                );
+            }
+            _chooseTargetCombo->setCurrentIndex(0);
+            _chooseTargetCombo->show();
+            break;
+        }
+        case openspace::HorizonsFile::ResultCode::UnknownError:
+            appendLog(
+                "Unknown error",
+                LogLevel::Error
+            );
+            if (!_latestHorizonsError.empty()) {
+                appendLog(fmt::format("Latest Horizons error: {}", _latestHorizonsError),
+                    LogLevel::Error
+                );
+            }
+            _errorMsg->setText("An unknown error occured");
+            return false;
```

Comment:
Yes, if the file is supposed to remain then it is return instead of break. That is the cases where we don't know what happened or could not find all information and give to the user directly. It is mostly a safe net for error cases that have not been discovered yet or don't know how to handle. Then the user can still go to the file and hopefully get an error message from there. Otherwise we would remove the file and the user would have no clue what went wrong.

Reply:
I tried to make this more clear in the code now so hopefully it wont be as confusing now.